src/Transports.AspNetCore/WebSockets/IWebSocketConnection.cs

```diff
@@ -0,0 +1,50 @@
+using GraphQL.Transport;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Represents a WebSocket connection, dispatching received messages over
+/// the connection to the specified <see cref="IOperationMessageProcessor"/>,
+/// and sending requested messages out the connection when requested.
+/// <br/><br/>
+/// <see cref="ExecuteAsync(IOperationMessageProcessor)"/> may be only called once.
+/// All members must be thread-safe.
+/// </summary>
+public interface IWebSocketConnection
+{
+    /// <summary>
+    /// Listens to incoming messages over the WebSocket connection,
+    /// dispatching the messages to the specified <paramref name="operationMessageProcessor"/>.
+    /// Returns or throws <see cref="OperationCanceledException"/> when the WebSocket connection is closed.
+    /// </summary>
+    Task ExecuteAsync(IOperationMessageProcessor operationMessageProcessor);
+
+    /// <summary>
+    /// Sends a message.
+    /// </summary>
+    Task SendMessageAsync(OperationMessage message);
+
+    /// <summary>
+    /// Closes the WebSocket connection.
+    /// </summary>
+    Task CloseConnectionAsync();
```

Comment:
```suggestion
    Task CloseAsync();
```

since the context is clear

Reply:
Also `CloseAsync` in interface looks excessive - there is always overload with two parameters. I would implement parameterless `CloseAsync()` method as extension method.

Reply:
> CloseAsync

I don't know.  Please re-review the updated comments and then confirm if you think we should go with `CloseAsync`.  This is a `IWebSocketConnection` after all, so having `Connection` in the name seems excessive, but I don't want to imply that disposal occurs (like `Close` for a `DbConnection`), which it does not.  On the other hand, it does (eventually) close the connection, so `CloseAsync` may be fine.

> extension method

Seems to me the normal closure message and/or behavior is dependent on the implementation.  I would not eliminate the method.

Reply:
After thinking about it for the last hour or so, I'll go with `CloseAsync` here; but please lmk if you have any other comments.